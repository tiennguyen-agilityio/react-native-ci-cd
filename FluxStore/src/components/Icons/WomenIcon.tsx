import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const WomenIcon = memo(
  ({
    width = 13,
    height = 20,
    color,
    disabled = false,
    isActive = false,
    style,
    onPress,
  }: IconProps) => {
    const {
      theme: {icon, text},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="women-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 13 20" fill="none">
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M6.25 1.25a5 5 0 100 10 5 5 0 000-10zM0 6.25a6.25 6.25 0 116.875 6.219v2.53h2.5a.625.625 0 010 1.25h-2.5v3.126a.625.625 0 11-1.25 0V16.25h-2.5a.625.625 0 110-1.25h2.5v-2.531A6.25 6.25 0 010 6.249z"
            fill={color ?? isActive ? text.light : icon.tertiary}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
