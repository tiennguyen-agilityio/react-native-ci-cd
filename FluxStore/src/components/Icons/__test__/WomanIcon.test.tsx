import {render, fireEvent, screen} from '@testing-library/react-native';

import {WomenIcon} from '../WomenIcon';

describe('WomenIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<WomenIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<WomenIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<WomenIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('women-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<WomenIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('women-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
