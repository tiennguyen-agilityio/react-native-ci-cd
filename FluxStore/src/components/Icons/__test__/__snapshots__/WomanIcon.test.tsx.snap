// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WomenIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="women-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={13}
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 13,
        },
      ]
    }
    vbHeight={20}
    vbWidth={13}
    width={13}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M6.25 1.25a5 5 0 100 10 5 5 0 000-10zM0 6.25a6.25 6.25 0 116.875 6.219v2.53h2.5a.625.625 0 010 1.25h-2.5v3.126a.625.625 0 11-1.25 0V16.25h-2.5a.625.625 0 110-1.25h2.5v-2.531A6.25 6.25 0 010 6.249z"
        fill={
          {
            "payload": 4288519581,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`WomenIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="women-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={13}
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 13,
        },
      ]
    }
    vbHeight={20}
    vbWidth={13}
    width={13}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M6.25 1.25a5 5 0 100 10 5 5 0 000-10zM0 6.25a6.25 6.25 0 116.875 6.219v2.53h2.5a.625.625 0 010 1.25h-2.5v3.126a.625.625 0 11-1.25 0V16.25h-2.5a.625.625 0 110-1.25h2.5v-2.531A6.25 6.25 0 010 6.249z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
